apps/emqx_gateway/test/emqx_coap_SUITE.erl

```diff
@@ -292,12 +292,37 @@ t_clients_get_subscription_api(_) ->
           end,
     with_connection(Fun).
 
+t_on_offline_event(_) ->
+    Fun = fun(Channel) ->
+                  application:start(emqx_modules),
+                  emqx_event_message_SUITE:load_config(),
+                  timer:sleep(100),
+
+                  emqx_event_message:enable(),
+                  timer:sleep(200),
+
+                  emqx_broker:subscribe(<<"$event/#">>),
```

Comment:
Plz meck `emqx_hooks:run/*` function to check whether the `client.disconnected` triggered.

The `$event/#` will be removed in the next release.